src/hooks/useNextClobPairId.ts

```diff
@@ -0,0 +1,118 @@
+import { useMemo } from 'react';
+import { useQuery } from 'react-query';
+
+import {
+  MsgCreateClobPair,
+  MsgCreateOracleMarket,
+  MsgCreatePerpetual,
+  MsgDelayMessage,
+  MsgUpdateClobPair,
+  TYPE_URL_MSG_CREATE_CLOB_PAIR,
+  TYPE_URL_MSG_CREATE_ORACLE_MARKET,
+  TYPE_URL_MSG_CREATE_PERPETUAL,
+  TYPE_URL_MSG_DELAY_MESSAGE,
+  TYPE_URL_MSG_UPDATE_CLOB_PAIR,
+} from '@dydxprotocol/v4-client-js';
+
+import type { PerpetualMarketResponse } from '@/constants/indexer';
+import { useDydxClient } from '@/hooks/useDydxClient';
+
+export const useNextClobPairId = () => {
+  const { isConnected, requestAllPerpetualMarkets, requestAllGovernanceProposals } =
+    useDydxClient();
+
+  const { data: perpetualMarkets, status: perpetualMarketsStatus } = useQuery({
+    enabled: isConnected,
+    queryKey: 'requestAllPerpetualMarkets',
+    queryFn: requestAllPerpetualMarkets,
+    refetchInterval: 60_000,
+    staleTime: 60_000,
+  });
+
+  const { data: allGovProposals, status: allGovProposalsStatus } = useQuery({
+    enabled: isConnected,
+    queryKey: 'requestAllActiveGovernanceProposals',
+    queryFn: () => requestAllGovernanceProposals(),
+    refetchInterval: 10_000,
+    staleTime: 10_000,
+  });
+
+  /**
+   *
+   * @param message from proposal. Each message is wrapped in a type any (on purpose).
+   * @param callback method used to compile all clobPairIds, perpetualIds, marketIds, etc.
+   */
+  const decodeMsgForClobPairId = (message: any, callback: (id?: number) => void): any => {
```

Comment:
Currently we do all our message decoding in v4-client. I think it would make sense to also move this to v4-client as well as a follow up.